src/crush/CrushCompiler.cc

```diff
@@ -816,6 +816,49 @@ void CrushCompiler::dump(iter_t const& i, int ind)
 }
 
 
+//Sometimes when defined the bucket above the item which the bucket
```

Comment:
@tchaikov : OK sir.How about  the example like below?
before adjust                      after adjust
root default {                     host localhost {
  id -1                                 id -2          
  item localhost weight 0.000
}                                   }

host localhost {                   root default {
  id -2                                 id -1          
                                       item localhost weight 0.000
}                                    }